cleanlab/filter.py

```diff
@@ -398,6 +421,171 @@ class 0, 1, ..., K-1.
+
+      frac_noise * number_of_mislabeled_examples_in_class_k.
+
+    num_to_remove_per_class : array_like
+      An iterable of length K, the number of classes.
+      E.g. if K = 3, ``num_to_remove_per_class=[5, 0, 1]`` would return
+      the indices of the 5 most likely mislabeled examples in class 0,
+      and the most likely mislabeled example in class 2.
+
+      Note
+      ----
+      Only set this parameter if ``filter_by='prune_by_class'``.
+      You may use with ``filter_by='prune_by_noise_rate'``, but
+      if ``num_to_remove_per_class=k``, then either k-1, k, or k+1
+      examples may be removed for any class due to rounding error. If you need
+      exactly 'k' examples removed from every class, you should use
+      ``filter_by='prune_by_class'``.
+
+    min_examples_per_class : int, default=1
+      Minimum number of examples per class to avoid flagging as label issues.
+      This is useful to avoid deleting too much data from one class
+      when pruning noisy examples in datasets with rare classes.
+
+    confident_joint : np.ndarray, optional
+      An array of shape ``(num_classes,K, K)`` representing the confident joint, the matrix used for identifying label issues, which
+      estimates a confident subset of the joint distribution of the noisy and true labels, ``P_{noisy label, true label}``.
+      Entry ``(j, k)`` in the matrix is the number of examples confidently counted into the pair of ``(noisy label=j, true label=k)`` classes.
+      The `confident_joint` can be computed using :py:func:`count.compute_confident_joint <cleanlab.count.compute_confident_joint>`.
+      If not provided, it is computed from the given (noisy) `labels` and `pred_probs`.
+
+    n_jobs : optional
+      Number of processing threads used by multiprocessing. Default ``None``
+      sets to the number of cores on your CPU.
+      Set this to 1 to *disable* parallel processing (if its causing issues).
+      Windows users may see a speed-up with ``n_jobs=1``.
+
+    verbose : optional
+      If ``True``, prints when multiprocessing happens.
+
+    Returns
+    -------
+    label_issues : np.ndarray
+      If `return_indices_ranked_by` left unspecified, returns a boolean **mask** for the entire dataset
+      where ``True`` represents a label issue and ``False`` represents an example that is
+      accurately labeled with high confidence.
+      If `return_indices_ranked_by` is specified, returns a shorter array of **indices** of examples identified to have
+      label issues (i.e. those indices where the mask would be ``True``), sorting by likelihood that the corresponding label is correct is not supported yet.
+
+      Note
+      ----
+      Obtain the *indices* of label issues in your dataset by setting
+      `return_indices_ranked_by`.
+
+    """
+    k = get_num_classes(
+        labels=labels, pred_probs=pred_probs, label_matrix=confident_joint, multi_label=True
+    )
+    num_classes = pred_probs.shape[1]
```

Comment:
isn't k same as num_classes, we're expecting pred_probs here, 
should multi_label work if pred_probs are None?

Reply:
add assert pred_probs is not None?